Revert "security: lsm_audit: add ioctl specific auditing"

This reverts commit c8c3cd48e44fe12a41cd20e46d36fcfe5a759fd7.

Bug: 22846070
Change-Id: Ibf79b4c76f22f0af2d39e377b551151837a782cd
Signed-off-by: Jeff Vander Stoep <jeffv@google.com>

diff --git a/include/linux/lsm_audit.h b/include/linux/lsm_audit.h
index ffb9c9d..1cc89e9 100644
--- a/include/linux/lsm_audit.h
+++ b/include/linux/lsm_audit.h
@@ -40,11 +40,6 @@
 	} fam;
 };
 
-struct lsm_ioctlop_audit {
-	struct path path;
-	u16 cmd;
-};
-
 /* Auxiliary data to use in generating the audit record. */
 struct common_audit_data {
 	char type;
@@ -58,7 +53,6 @@
 #define LSM_AUDIT_DATA_KMOD	8
 #define LSM_AUDIT_DATA_INODE	9
 #define LSM_AUDIT_DATA_DENTRY	10
-#define LSM_AUDIT_DATA_IOCTL_OP	11
 	union 	{
 		struct path path;
 		struct dentry *dentry;
@@ -74,7 +68,6 @@
 		} key_struct;
 #endif
 		char *kmod_name;
-		struct lsm_ioctlop_audit *op;
 	} u;
 	/* this union contains LSM specific data */
 	union {

diff --git a/security/lsm_audit.c b/security/lsm_audit.c
index 8f6b8e8..8d8d97d 100644
--- a/security/lsm_audit.c
+++ b/security/lsm_audit.c
@@ -245,21 +245,6 @@
 		}
 		break;
 	}
-	case LSM_AUDIT_DATA_IOCTL_OP: {
-		struct inode *inode;
-
-		audit_log_d_path(ab, " path=", &a->u.op->path);
-
-		inode = a->u.op->path.dentry->d_inode;
-		if (inode) {
-			audit_log_format(ab, " dev=");
-			audit_log_untrustedstring(ab, inode->i_sb->s_id);
-			audit_log_format(ab, " ino=%lu", inode->i_ino);
-		}
-
-		audit_log_format(ab, " ioctlcmd=%hx", a->u.op->cmd);
-		break;
-	}
 	case LSM_AUDIT_DATA_DENTRY: {
 		struct inode *inode;
 
